### PR TITLE
Issue/add share site

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Add an action to share sites in the Reader for blogs and feeds.  The share site behavior mimics the share post behavior.  See https://github.com/wordpress-mobile/WordPress-iOS/pull/10899 for details about the iOS counterpart changes.  See the screenshots below for illustration.

share_site

### Test
1. Go to ***Reader*** tab.
2. Tap site button in header of post.
3. Notice ***Share*** action in top app bar.
4. Tap ***Share*** action in top app bar.
5. Notice share action sheet is shown.
6. Tap email app in list.
7. Notice blog/feed name in email subject and blog/feed URL in email content.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
The `RELEASE-NOTES.txt` document was updated in 9c87a14 with the following:
> Added an action to share sites in the Reader

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListActivity.java [hunk 1]
```diff
@@ -67,8 +75,10 @@ public void onClick(View view) {
                     long feedId = getIntent().getLongExtra(ReaderConstants.ARG_FEED_ID, 0);
                     if (feedId != 0) {
                         showListFragmentForFeed(feedId);
+                        mSiteId = feedId;
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListActivity.java]: We also need to define `mSiteId` when `savedInstanceState` is not `null`, like after configuration change.

Comment (theck13) [reply WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostListActivity.java]: Yep!  I added variable restoration in 124e030.

**Comment by mattmiklic** [comment]
Makes sense to re-use the same pattern used for sharing posts. Icon placement is good, don't see any other issues here. Looks good to me. 👍

**Comment by theck13** [comment]
I'll go ahead and merge this since it has passed code and design review.

**Pull request merged**